src/Bicep.Core.Samples/Files/Completions/declarations.json

```diff
@@ -167,6 +167,204 @@
       "newText": "param ${1:Identifier} string {\n  secure: true\n}"
     }
   },
+  {
+    "label": "res-aks-cluster",
+    "kind": "snippet",
+    "detail": "Kubernetes Service Cluster",
+    "documentation": {
+      "kind": "markdown",
+      "value": "```bicep\nresource aksCluster1 'Microsoft.ContainerService/managedClusters@2020-02-01' = {\n  name: 'aksCluster1'\n  location: resourceGroup().location\n  properties: {\n    kubernetesVersion: '${2|1.15.7,1.15.5,1.14.8|}'\n    dnsPrefix: 'dnsprefix'\n    agentPoolProfiles: [\n      {\n        name: 'agentpool'\n        count: 2\n        vmSize: 'Standard_A1'\n        osType: 'Linux'\n      }\n    ]\n    linuxProfile: {\n      adminUsername: 'adminUserName'\n      ssh: {\n        publicKeys: [\n          {\n            keyData: 'keyData'\n          }\n        ]\n      }\n    }\n    servicePrincipalProfile: {\n      clientId: 'servicePrincipalAppId'\n      secret: 'servicePrincipalAppPassword'\n    }\n  }\n}\n```"
+    },
+    "deprecated": false,
+    "preselect": false,
+    "sortText": "2_res-aks-cluster",
+    "insertTextFormat": "snippet",
+    "insertTextMode": "asIs",
+    "textEdit": {
+      "range": {},
+      "newText": "resource aksCluster1 'Microsoft.ContainerService/managedClusters@2020-02-01' = {\n  name: '${1:aksCluster1}'\n  location: resourceGroup().location\n  properties: {\n    kubernetesVersion: '${2|1.15.7,1.15.5,1.14.8|}'\n    dnsPrefix: '${3:dnsprefix}'\n    agentPoolProfiles: [\n      {\n        name: 'agentpool'\n        count: ${4:2}\n        vmSize: '${5:Standard_A1}'\n        osType: 'Linux'\n      }\n    ]\n    linuxProfile: {\n      adminUsername: '${7:adminUserName}'\n      ssh: {\n        publicKeys: [\n          {\n            keyData: '${8:keyData}'\n          }\n        ]\n      }\n    }\n    servicePrincipalProfile: {\n      clientId: '${9:servicePrincipalAppId}'\n      secret: '${10:servicePrincipalAppPassword}'\n    }\n  }\n}"
```

Comment:
For all snippets, the symbolic name should use the same placeholder as the name.

Reply:
Okay, I didn't make any changes to decompiled files. I noticed the placeholders/names have trailing 1. So should we retain it e.g. call it aksCluster1 or aksCluster is fine?

Reply:
Personally I'd remove the trailing 1.

Reply:
But should we turn it into a named snippet placeholder on all of the snippets?

Reply:
Updated here:
https://github.com/Azure/bicep/commit/e74b8da5cc561cc934a1b5c21335071ba3594efd

@majastrz , can symbolic name have placeholders?

I see this error:
![image](https://user-images.githubusercontent.com/30270536/111423500-a6c00b80-86ad-11eb-8271-b1bfef7a7ce7.png)

Am I missing something?

Reply:
That's because symbolic names don't allow quotes.

Reply:
This is where we need to get creative.  This is also a question for another PR (and maybe work item), but my current thoughts...

> can symbolic name have placeholders?

That question doesn't make sense - a snippet is just a string that contains placeholders.  Using bicep files to contain the snippets is just our convention for ease of authoring.

Ideally the format we use in the snippet bicep files should should few errors in the issues list.  Right now we see this in the snippet bicep files:

![image](https://user-images.githubusercontent.com/6913354/111509946-7c506b80-870a-11eb-91b1-26f22e356f8f.png)

We could allow some additional notations in the snippet files that conform better to bicep.  E.g. something along these lines:
```bicep
resource PLACEHOLDER1_automationAccount1 'Microsoft.Automation/automationAccounts@2015-10-31' = {
  name: 'PLACEHOLDER1_automationAccount1'
  location: resourceGroup().location
  tags: {}
  properties: {
    sku: {
      name: '\${2|Free,Basic|}'
    }
  }
}
```
Unfortunately even with escaping the "$" in name, we still get a validation error:
![image](https://user-images.githubusercontent.com/6913354/111510945-7ad37300-870b-11eb-9aaa-53a5acb11a77.png)
But it's a lot closer.

Reply:
@StephenWeatherford This would mean doing work in the language server to convert snippets because of testing, which doesn't seem right to me. Take a look at @anthony-c-martin's suggestion below. With that approach, we would mirror the standard snippet lifecycle of: insert snippet with placeholders -> replace placeholders -> compile.

Reply:
This has nothing to do with testing. We want the resource name to be a placeholder, right?  That means the following in the snippet bicep file:
```bicep
resource ${1:automationAccount} 'Microsoft.Automation/automationAccounts@2015-10-31' = {
  name: '${1:automationAccount}'
  ...
```
But this is invalid bicep:
```
resource ${1:automationAccount} 
```
Thus when you're authoring the snippet bicep file, you get errors.

Am I missing something?

Reply:
I am going to go ahead and merge the changes. I'll look into this before I send out PR with next set of changes. Thanks!